docs: clarify update-artifact wording (DOCS-1215) (#2329)
wandb/docs: models/artifacts/update-an-artifact.mdx

---
description: Update an existing artifact while a run is active or using only the Public API.
title: Update an artifact
---

> Pass desired values to update the `description`, `metadata`, and `alias` of an artifact. Call the `save()` method to persist changes on the W&B servers. You can do this while a run is active, or with [`wandb.Api`](/models/ref/python/public-api/api) without calling `wandb.init()`.

Reviewer: ```suggestion
Pass desired values to update the `description`, `metadata`, and `alias` of an artifact. Update a run previously logged to W&B with the W&B Public API with ([`wandb.Api`](/models/ref/python/public-api/api)). Use `wandb.Run.save()` to update an artifact when is first initialized and still active. 
```

<Note>

> **When to use Artifact.save() or wandb.Run.log_artifact()**

Reviewer: ```suggestion
**When to use wandb.Artifact.save() or wandb.Run.log_artifact()**
```

- Use `Artifact.save()` to update an existing artifact without starting a new run.
- Use `wandb.Run.log_artifact()` to create a new artifact and associate it with a specific run.
</Note>

> Use the W&B Public API ([`wandb.Api`](/models/ref/python/public-api/api)) to update an artifact without calling `wandb.init()`. Use the Artifact API ([`wandb.Artifact`](/models/ref/python/experiments/artifact)) while a run is active.

Reviewer: ```suggestion
Use the W&B Public API ([`wandb.Api`](/models/ref/python/public-api/api)) to update an artifact. Use the wandb.Artifact ([`wandb.Artifact`](/models/ref/python/experiments/artifact)) Class while a run is active.
```

Reviewer: We might be able to omit this sentence as we state this in the opening paragraph.

    artifact.save()
```
</Tab>

> <Tab title="Public API only">

Reviewer: ```suggestion
<Tab title="W&B Public API">
```

> The following example updates an artifact with [`wandb.Api`](/models/ref/python/public-api/api) and no active run:

Reviewer: ```suggestion
The following example updates an artifact with [`wandb.Api`](/models/ref/python/public-api/api):
```
